File: opensource/data/src/main/java/com/sos/scheduler/engine/data/job/TaskId.java
package com.sos.scheduler.engine.data.job;

import com.sos.scheduler.engine.data.base.StringValue;
import org.codehaus.jackson.map.annotate.JsonDeserialize;

@JsonDeserialize(using=TaskIdDeserializer.class)
public class TaskId extends StringValue {
	public TaskId(int x) { super( String.valueOf(x) ); }
}

File: opensource/data/src/main/scala/com/sos/scheduler/engine/data/job/TaskId.scala
package com.sos.scheduler.engine.data.job

import com.fasterxml.jackson.annotation.{JsonCreator, JsonValue}
import com.sos.scheduler.engine.cplusplus.runtime.annotation.ForCpp
import scala.annotation.meta.getter

@ForCpp
final case class TaskId(
  @(JsonValue @getter) @(ForCpp @getter) value: Int) {

  override def toString = string

  def string = value.toString
}

object TaskId {
  @JsonCreator def jsonCreator(taskId: Int) =
    new TaskId(taskId)
}
